scripts/train_alternating.py

```
@@ -0,0 +1,283 @@
+    if task_type == "node":
+        graph_pred = out.get("graph_pred_spice_energy")
+    else:
+        graph_pred = out.get("graph_pred_main")
+    
+    node_loss = torch.tensor(0.0, device=device)
+    if node_pred is not None and hasattr(batch, 'node_y') and batch.node_y is not None:
+        if batch.node_y.numel() > 0 and node_pred.numel() > 0:
+            if node_pred.shape == batch.node_y.shape:
+                node_loss = nn.MSELoss()(node_pred, batch.node_y)
+            else:
+                logger.warning(f"Node prediction shape {node_pred.shape} mismatch with target shape {batch.node_y.shape}. Skipping node loss.")
```

Ruff (0.11.9)

Undefined name `logger`

(F821)